src/container.ts

```diff
@@ -573,10 +584,41 @@ export function createContainer<T extends object = any, U extends object = any>(
    *
    * @see lib/loadModules.js documentation.
    */
-  function loadModules(
+  function loadModules<ESM extends boolean = false>(
+    globPatterns: Array<string | GlobWithOptions>,
+    opts: LoadModulesOptions<ESM>
+  ): Promise<AwilixContainer> | AwilixContainer {
+    const _loadModulesDeps = {
+      require:
+        options!.require ||
+        function (uri) {
+          return require(uri)
+        },
+      listModules,
+      container,
+    }
+    if (opts?.esModules) {
+      return new Promise((resolvePromise, reject) => {
+        return (realLoadModules(
+          _loadModulesDeps,
+          globPatterns,
+          opts
+        ) as Promise<LoadModulesResult>)
+          .then(() => {
+            resolvePromise(container)
+          })
+          .catch(reject)
+      })
+    } else {
+      realLoadModules(_loadModulesDeps, globPatterns, opts)
+      return container
+    }
+  }
+
+  async function loadModulesNative(
```

Comment:
Let's keep the API simple by sticking with a single `loadModules` that takes the options arg.

Reply:
Sorry, this was a remain from one of my experiments trying to please TypeScript, I'll remove it.